packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1814,7 +1818,7 @@ class _TabBarState extends State<TabBar> {
 
         final Color dividerColor = widget.dividerColor ?? tabBarTheme.dividerColor ?? _defaults.dividerColor!;
         final double dividerHeight = widget.dividerHeight ?? tabBarTheme.dividerHeight ?? _defaults.dividerHeight!;
-        final bool showDivider = dividerColor != Colors.transparent && dividerHeight > 0;
+        final bool showDivider = dividerHeight > 0;
```

Comment:
Then the first comment you added above is not true, right? The divider will still be drawn even when it is transparent.